### PR TITLE
Add Mesos Marathon application module for being managable with Ansible

### PR DESCRIPTION
**Pull request by signal-09**
- New Module Pull Request

##### COMPONENT NAME
marathon_app

```yaml
- name: Create a Marathon application
  marathon_app:
    marathon_uri: http://marathon.mesos:8080
    inline_data:
      id: /foo
      instances: 5
      cmd: "sleep infinity"
      cpus: 0.1
      mem: 16
    state: present
```

##### ANSIBLE VERSION
```
ansible 2.4.0
```

##### ADDITIONAL INFORMATION
This module provide a way to manage Marathon applications inside a Mesos cluster through Ansible.
The module use the [Marathon REST API](http://mesosphere.github.io/marathon/api-console/index.html)

```yaml
- name: Scale a Marathon application
  marathon_app:
    url: https://admin:secret@marathon.mesos:8443
    inline_data:
      id: /foo
      instances: 10
    state: update
    wait: yes
```

### CODE REVIEW

File: lib/ansible/modules/clustering/mesos/marathon_app.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                module.fail_json(msg="No valid data could be found.")
+        except:
+            module.fail_json(msg="The file '%s' was not found or contained invalid JSON data" % file_reference)
+
+    marathon = MarathonApp(module)
+
+    if module.params['state'] in ['absent', 'replace']:
+        body = marathon.delete_app(data)
+        marathon.results.append(body)
+    if module.params['state'] in ['present', 'update', 'replace']:
```
Comment (resmo) [inline lib/ansible/modules/clustering/mesos/marathon_app.py]: Why so many states? 

state=replace should be removed. in state=present, the code should handle a replacement inline (get to the state defined). Is this possible?

File: lib/ansible/modules/clustering/mesos/marathon_app.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+            data = json.load(f)
+            f.close()
+            if not data:
+                module.fail_json(msg="No valid data could be found.")
+        except:
+            module.fail_json(msg="The file '%s' was not found or contained invalid JSON data" % file_reference)
+
+    marathon = MarathonApp(module)
+
+    if module.params['state'] in ['absent', 'replace']:
```
Comment (resmo) [inline lib/ansible/modules/clustering/mesos/marathon_app.py]: why 2 states?

File: lib/ansible/modules/clustering/mesos/marathon_app.py (hunk 1 above)
Comment (signal-09) [reply lib/ansible/modules/clustering/mesos/marathon_app.py]: I can merge state=present with state=update, at the moment they do the same thing: change elements if defined and preserve the existing ones. state=replace force to "reset" values:
```yaml
- name: Create App
  marathon_app:
    inline_data:
      id: /foo
      instances: 1
      cmd: sleep infinity
      cpus: 0.1
      mem: 16
    state: present
```
With sate=present (or state=update), unspecified attributes are preserved (cmd, mem, cpus):
```yaml
- name: Scale App
  marathon_app:
    inline_data:
      id: /foo
      instances: 10
    state: present
```

The state=replace guarantee that unspecified attributed will be deleted or set with default values (instances=1):
```yaml
- name: Reset App
  marathon_app:
    inline_data:
      id: /foo
      cmd: sleep infinity
      mem: 10
      cpus: 0.1
    state: replace
```

**Comment by Oloremo** [comment]
any updates?

**Comment by Akasurde** [comment]
Please submit a new module request to https://github.com/ansible-collections/community.general. Thanks for your contribution.

**Pull request closed without merging**